packages/idealo.design/src/server/server.js

```diff
@@ -111,15 +115,12 @@ if (CLIENT_ID) {
   }));
 }
 
-process.env.dangerousTestModeArgument = 'false';
-
 function isAuthenticated(req, res, next) {
-  if(req.session.user){
-    return next();
-  }
-  if(process.env.dangerousTestModeArgument==='true'){
+  const testUser = req.header('authorization')
+  if(req.session.user || testUser){
```

Comment:
much better 👍  

maybe you can even supply the `req.session.user` in the request filter in the provider test? If you manage to do that, you can leave the isAuthenticated function as simple as it was before :)